universal-application-tool-0.0.1/app/auth/UATProfile.java

```diff
@@ -1,11 +1,46 @@
 package auth;
 
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import models.Account;
+import models.Applicant;
 import org.pac4j.core.profile.CommonProfile;
 
 public class UATProfile extends CommonProfile {
+  private final Clock clock;
+  private Applicant applicant;
+
   public UATProfile() {
+    this(Clock.systemDefaultZone());
+  }
+
+  public UATProfile(Clock clock) {
     super();
-    // TODO: write this to the database, read the ID back, and use that as the ID here.
-    setId("TODO");
+    this.clock = clock;
+  }
+
+  public Applicant getApplicant() {
+    if (applicant == null) {
+      this.applicant = new Applicant();
+      this.applicant.id = Long.valueOf(this.getId());
+      this.applicant.refresh();
+    }
+    return applicant;
+  }
+
+  public Account getAccount() {
+    return getApplicant().getAccount();
+  }
+
+  public void init() {
```

Comment:
suggestion (non-blocking): Consider adding a docstring to this, mostly for how to use it. One of my questions is: Does `init` always need to be called immediately after the constructor? If yes, why doesn't this logic live in the constructor? If no, how do I decide when to call `init`?

Reply:
Sure.  The rule is, no work can be done in constructors, so `init` is "the part of the constructor that does work", basically.